web/setups/submitissues.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Copyright (c) 2017 University of Illinois, NCSA.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the
+ * University of Illinois/NCSA Open Source License
+ * which accompanies this distribution, and is available at
+ * http://opensource.ncsa.illinois.edu/license.html
+ */
+
+// bug reporting page page
+
+$title = $_POST['title'];
+$description = $_POST['description'];
+$username = $_POST['username'];
+$password = $_POST['password'];
+
+//flags
+$isCurlFailed = FALSE;
+
+include 'page.template.php';
+echo "<h1> Report the bug</h1>";
+
+if (isset($title) && !empty($title) &&
+    isset($description) && !empty($description) &&
+    isset($username) && !empty($username) &&
+    isset($password) && !empty($password)) {
+
+      $service_url = "https://api.github.com/repos/PecanProject/pecan/issues";
+
+      $user_agent = "Pecan-application";
+      $curl = curl_init($service_url);
+      
+      $curl_post_data = array ('title' => $title,
+                               'body' => $description,
+                               'lables' => 'Bug' );
+
+      $data_string = json_encode($curl_post_data);
+
+      // setting curl to do a POST request
+      curl_setopt($curl, CURLOPT_USERPWD, $username.":".$password);   // authentication
+      curl_setopt($curl, CURLOPT_USERAGENT, $user_agent);
+      curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+      curl_setopt($curl, CURLOPT_POST, true);
+      curl_setopt($curl, CURLOPT_POSTFIELDS, $data_string);
+      curl_setopt($curl, CURLOPT_HTTPHEADER, array(
+        'Content-Type: application/json',
+        'Content-Length: ' . strlen($data_string))
+      );
+
+      // execute the curl request
+      $curl_response = curl_exec($curl);
+
+      $decode =  json_decode($curl_response);
+
+      //var_dump($decode);
+      if (curl_getinfo ($curl, CURLINFO_HTTP_CODE) == 201)
+      {
+// Generate tha page to inform the user
+?>
+        <div class="alert alert-success" role="alert">
+          Bug Report Sucessfully submitted
+        </div>
+        <div class="alert alert-info" role="alert">
+          You can refer it by visiting the following link
+          <a href="<?php echo $decode->html_url; ?>"> <?php echo $decode->html_url; ?></a>
+        </div>
+
+        <div class="alert alert-info" role="alert">
+          You can submit another one
+        </div>
+
+<?php
+      }else {
+?>
+        <div class="alert alert-danger" role="alert">Failed to submit the Bug Report</div>
+        <div class="alert alert-warning" role="alert">Please check your github credentials</div>
```

Comment:
Does this mean one has to have github credentials to submit a bug report? If that's the case, what's the advantage of this page over the current practice of just linking to "New Issue" on github itself?